### PR TITLE
Bias variance decomposition formula

### PR DESCRIPTION
**Pull request by JiaZhang42**
The expectation should be inside the square.
Plus, I think the last line directly follows the second line. The expansion in the 3rd and 4th lines seems unnecessary.

### CODE REVIEW

File: python/factor-selection-via-machine-learning.qmd [hunk 1]
```diff
@@ -39,7 +39,7 @@ MSE &=E\left(\left(y-\hat{f}(x)\right)^2\right)=E\left(\left(f(x)+\epsilon-\hat{
 &= \underbrace{E\left(\left(f(x)-\hat{f}(x)\right)^2\right)}_{\text{total quadratic error}}+\underbrace{E\left(\epsilon^2\right)}_{\text{irreducible error}} \\
 &= E\left(\hat{f}(x)^2\right)+E\left(f(x)^2\right)-2E\left(f(x)\hat{f}(x)\right)+\sigma_\varepsilon^2\\
 &=E\left(\hat{f}(x)^2\right)+f(x)^2-2f(x)E\left(\hat{f}(x)\right)+\sigma_\varepsilon^2\\
-&=\underbrace{\text{Var}\left(\hat{f}(x)\right)}_{\text{variance of model}}+ \underbrace{E\left((f(x)-\hat{f}(x))\right)^2}_{\text{squared bias}} +\sigma_\varepsilon^2. 
+&=\underbrace{\text{Var}\left(\hat{f}(x)\right)}_{\text{variance of model}}+ \underbrace{\left(E(f(x)-\hat{f}(x))\right)^2}_{\text{squared bias}} +\sigma_\varepsilon^2.
```
Comment (voigtstefan) [inline python/factor-selection-via-machine-learning.qmd:42]: correct, thanks for pointing out this typo

File: r/factor-selection-via-machine-learning.qmd [hunk 2]
```diff
@@ -36,7 +36,7 @@ MSE &=E((y-\hat{f}(x))^2)=E((f(x)+\epsilon-\hat{f}(x))^2)\\
 &= \underbrace{E((f(x)-\hat{f}(x))^2)}_{\text{total quadratic error}}+\underbrace{E(\epsilon^2)}_{\text{irreducible error}} \\
 &= E\left(\hat{f}(x)^2\right)+E\left(f(x)^2\right)-2E\left(f(x)\hat{f}(x)\right)+\sigma_\varepsilon^2\\
 &=E\left(\hat{f}(x)^2\right)+f(x)^2-2f(x)E\left(\hat{f}(x)\right)+\sigma_\varepsilon^2\\
-&=\underbrace{\text{Var}\left(\hat{f}(x)\right)}_{\text{variance of model}}+ \underbrace{E\left((f(x)-\hat{f}(x))\right)^2}_{\text{squared bias}} +\sigma_\varepsilon^2. 
+&=\underbrace{\text{Var}\left(\hat{f}(x)\right)}_{\text{variance of model}}+ \underbrace{\left(E(f(x)-\hat{f}(x))\right)^2}_{\text{squared bias}} +\sigma_\varepsilon^2.
```
Comment (voigtstefan) [inline r/factor-selection-via-machine-learning.qmd:39]: correct, thanks for pointing out this typo

File: python/factor-selection-via-machine-learning.qmd [hunk 3]
```diff
@@ -366,7 +366,7 @@ Cross-validation is a technique that allows us to alleviate this problem. We app
...
 
-For our sample, we consider a time-series cross-validation sample. This means that we tune our models with 20 random samples of length five years with a validation period of four years. For a grid of possible hyperparameters, we then fit the model for each fold and evaluate $\hat{\text{MSPE}}$ in the corresponding validation set. Finally, we select the model specification with the lowest MSPE in the validation set. First, we define the cross-validation folds based on our training data only.
+For our sample, we consider a time-series cross-validation sample. This means that we tune our models with 20 samples of length five years with a validation period of four years. For a grid of possible hyperparameters, we then fit the model for each fold and evaluate $\hat{\text{MSPE}}$ in the corresponding validation set. Finally, we select the model specification with the lowest MSPE in the validation set. First, we define the cross-validation folds based on our training data only.
```
Comment (voigtstefan) [inline python/factor-selection-via-machine-learning.qmd:369]: I'm not sure if these are semantics, but in this specific case with time-series cross-validation, I actually do not see randomness. The cross-validation split determines which observation belongs to which fold, so I'd instead not refer to a random sample in this context. (Same for the Python version)

**Review by voigtstefan — CHANGES_REQUESTED** [review]
Thanks for the comments! I did not get to test the `augment` proposal yet, but will do as soon as possible.

File: python/factor-selection-via-machine-learning.qmd (hunk 3 above)
Comment (christophscheuch) [reply python/factor-selection-via-machine-learning.qmd:369]: The proposal is to remove the "random", so I interpret your statement as agreement :)

Comment (voigtstefan) [reply python/factor-selection-via-machine-learning.qmd:369]: Oh, never mind, I was too sloppy :-D 
Sure, agree 👍

File: r/factor-selection-via-machine-learning.qmd [hunk 4]
```diff
@@ -250,8 +250,7 @@ First, we focus on the predicted values $\hat{y}_t = x_t\hat\beta^\text{Lasso}.$
 ```{r}
 predicted_values <- lm_fit |>
   fit(data = training(split)) |>
-  predict(data |> filter(industry == "manuf")) |>
-  bind_cols(data |> filter(industry == "manuf")) |>
+  augment(data |> filter(industry == "manuf")) |>
```
Comment (voigtstefan) [inline r/factor-selection-via-machine-learning.qmd:253]: works, approved 👍

File: python/factor-selection-via-machine-learning.qmd (hunk 1 above)
Comment (JiaZhang42) [reply python/factor-selection-via-machine-learning.qmd:42]: thanks

File: r/factor-selection-via-machine-learning.qmd (hunk 2 above)
Comment (JiaZhang42) [reply r/factor-selection-via-machine-learning.qmd:39]: thanks

**Pull request merged**